tests/v0_tests/test_custom_vector_search.py

```diff
@@ -75,27 +79,67 @@ def test_custom_search_results(self):
+                "tensorss": [
+                    {"vector": [1, ] * 512, "weight": 0},
+                    {"vector": [2, ] * 512, "weight": 0}
+                ], 
+            })
             raise AssertionError
         except MarqoWebError:
             pass
 
-    def test_custom_context_with_wrong_dimension(self):
-        correct_dimension = 512
-        wrong_dimension = 200
-        if self.IS_MULTI_INSTANCE:
-            self.warm_request(self.client.index(self.index_name_1).search, q=self.query,
-                              context = {"tensor": [{"vector": [1, ] * correct_dimension, "weight": 0}, {"vector": [2, ] * correct_dimension, "weight": 0}], })
+    def test_context_dimension_have_different_dimensions_to_index(self):
+         correct_context = {"tensor": [{"vector": [1, ] * 384, "weight": 1}]}
+         wrong_context = {"tensor": [{"vector": [1, ] * 2, "weight": 1}]}
+         if self.IS_MULTI_INSTANCE:
+             self.warm_request(lambda _: self.search_with_context(correct_context))
+         try:
+             self.search_with_context(wrong_context)
+             raise AssertionError
+         except MarqoWebError as e:
+            assert "The provided vectors are not in the same dimension of the index" in str(e)
+
+    def test_context_dimension_have_inconsistent_dimensions(self):
+         correct_context = {"tensor": [{"vector": [1, ] * 384, "weight": 1}, {"vector": [2, ] * 384, "weight": 0}]}
+         wrong_context = {"tensor": [{"vector": [1, ] * 384, "weight": 1}, {"vector": [2, ] * 385, "weight": 0}]}
+         if self.IS_MULTI_INSTANCE:
+             self.warm_request(lambda _: self.search_with_context(correct_context))
+         try:
+             self.search_with_context(wrong_context)
+             raise AssertionError
+         except MarqoWebError as e:
+            assert "The provided vectors are not in the same dimension of the index" in str(e)
+
+    def test_context_vector_with_flat_query(self):
+        self.query = "What are the best pets"
+        context = {"tensor": [{"vector": [1, ] * 384, "weight": 1}, {"vector": [2, ] * 384, "weight": 0}]}
         try:
-            custom_res = self.client.index(self.index_name_1).search(q=self.query,
-                    context = {"tensor": [{"vector": [1, ] * wrong_dimension, "weight": 0}, {"vector": [2, ] * 512, "weight": 0}], })
-            raise AssertionError
+            result = self.search_with_context(context)
+            raise AssertionError(f"The query should not be accepted. Returned: {result}")
         except MarqoWebError as e:
-            assert "The provided vectors are not in the same dimension of the index" in str(e)
+            assert "This is not supported as the context only works when the query is a dictionary." in str(e)
+        finally:
+
+            ## Ensure other tests are not affected
+            self.query = {"What are the best pets": 1}
+
+class TestCustomBulkVectorSearch(TestCustomVectorSearch):
+
```

Comment:
This doesn't seem to be fully implemented

Reply:
What do you mean?